### PR TITLE
Deprecate Nodes or Relationship use on the RHS of a Set Clause

### PR DESCRIPTION
**Pull request by gem-neo4j**

### CODE REVIEW

**Comment by gem-neo4j** [comment]
[Deprecate nodes/rels in RHS of SET](https://trello.com/c/JSyKWlgG/61-deprecate-nodes-rels-in-rhs-of-set)

**Review by AzuObs — APPROVED** [review]
LGTM!

File: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc [hunk 1]
```diff
@@ -422,6 +422,29 @@ WHERE NOT isEmpty([1, 2, 3])
...
+a|
+label:syntax[]
+label:deprecated[]
+[source, cypher, role="noheader"]
+----
+MATCH (n)-[r:REL]->(m) SET n=r
+----
+a|
+Use the properties() function instead to get the map of properties of nodes/relationships that
+can then be used in a SET clause:
```
Comment (nvitucci) [inline cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc]:
```suggestion
Use the `properties()` function instead to get the map of properties of nodes/relationships that
can then be used in a `SET` clause:
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SetTest.scala [hunk 2]
```diff
@@ -123,12 +123,12 @@ class SetTest extends DocumentingTest with QueryStatisticsTestSupport {
       }
     }
     section("Copy properties between nodes and relationships", "set-copying-properties-between-nodes-and-relationships") {
-      p("""`SET` can be used to copy all properties from one node or relationship to another.
+      p("""`SET` can be used to copy all properties from one node or relationship to another using the function properties(),
```
Comment (nvitucci) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SetTest.scala]:
```suggestion
      p("""`SET` can be used to copy all properties from one node or relationship to another using the `properties()` function,
```

**Review by nvitucci — APPROVED** [review]

**Pull request merged**